src/main/java/com/exasol/adapter/dialects/sqlserver/SqlServerIdentifier.java

```diff
@@ -0,0 +1,62 @@
+package com.exasol.adapter.dialects.sqlserver;
+
+import java.util.Objects;
+
+import com.exasol.db.Identifier;
+
+/**
+ * Represents an identifier in the Sql Server database.
+ */
+public class SqlServerIdentifier implements Identifier {
+    private final String id;
+
+    private SqlServerIdentifier(final String id) {
+        this.id = id;
+    }
+
+    /**
+     * Get the quoted identifier as a {@link String}.
+     *
+     * @return quoted identifier
+     */
+    @Override
+    public String quote() {
+        return "[" + this.id + "]";
+    }
+
+    /**
+     * Create a new {@link SqlServerIdentifier}.
+     *
+     * @param id the identifier as {@link String}
+     * @return new {@link SqlServerIdentifier} instance
+     */
+    public static SqlServerIdentifier of(final String id) {
+        if (validate(id)) {
+            return new SqlServerIdentifier(id);
+        } else {
+            throw new AssertionError("E-ID-3: Unable to create identifier \"" + id //
+                    + "\" because it contains illegal characters." //
+                    + " For information about valid identifiers, please refer to" //
+                    + " https://docs.microsoft.com/en-us/sql/relational-databases/databases/database-identifiers?redirectedfrom=MSDN&view=sql-server-ver15");
+        }
+    }
+
+    private static boolean validate(final String id) {
+        return !id.contains("[") && !id.contains("]");
```

Comment:
Not sure, what you mean. But on the other hand we might also link to https://docs.microsoft.com/de-de/sql/t-sql/statements/set-quoted-identifier-transact-sql?view=sql-server-ver15 because I guess that's the reason why we don't use `"` for these identifiers. I think the documentation doesn't even disallow `[` within `[]`, but also doesn't say that it's allowed.

Reply:
I would forbid [] for because I don't see any other fast way to validate this. Otherwise, I'd need to parse the string to count the [] pairs

Reply:
Please change both links back to https://docs.microsoft.com/de-de/sql/relational-databases/databases/database-identifiers?view=sql-server-ver15 because the link I mentioned is not what we use.